include/boost/url/impl/url_view.ipp

```diff
@@ -442,6 +444,51 @@ encoded_fragment() const noexcept
     return s.substr(1);
 }
 
+//------------------------------------------------
+//
+// Comparisons
+//
+//------------------------------------------------
+
+int
+url_view::
+compare(const url_view& other) const noexcept
+{
+    int comp = detail::icompare(
+        scheme(),
+        other.scheme());
+    if ( comp != 0 )
+        return comp;
+
+    comp = detail::pct_decode_icompare_unchecked(
+        encoded_authority(),
+        other.encoded_authority());
+    if ( comp != 0 )
+        return comp;
+
+    comp = detail::remove_dot_segments_compare(
```

Comment:
wouldn't this be more appropriately named `compare_encoded_path`

Reply:
it is an implementation detail though, so the requirements for names can be relaxed a little bit. If it was public, that'd be another story.